### DISCUSSION TITLE
Release 1.1

### DISCUSSION THREAD

**Discussion by Noordfrees** [widelands/widelands#5352] [General]
It's been 10 months since the release of Widelands v1.0, and in my opinion we should start thinking about the next **release 1.1** soonish to uphold the target cycle of one release per annum. This time last year we were already in feature freeze.

Although I have the impression that development has slowed down somewhat over the last two release cycles, and several long-standing wishlist items are to be postponed yet again, we have a good number of new features and bugfixes that warrant a new release. And there are several more major goodies currently in review, such as two new scenarios. In my opinion there are no showstopper bugs now known, though some of the open crash reports still need to be investigated.

I propose to set the date for the **Widelands 1.1 First Snow Feature Freeze** to **Sunday, the First of May 2022**. This will give us enough time to add more features and fix important bugs, and of course branches deemed release-critical are not subject to the freeze. I will go through the list of issues soonish and retarget what is not urgent.

Do you agree? Or should we allow more time for more new features? Are there any specific enhancements/bugfixes/tweaks you'd like to see in v1.1? Opinions, as always, welcome :)

**Comment by frankystone** [comment]
+1

**Comment by matthiakl** [comment]
+1
Maybe this will motivate me to return to the code for some bugfixes.

**Comment by tothxa** [comment]
+1, but:

Where do high resolution re-exported graphics of old buildings and workers fit in? Should they be submitted at least as a PR until 1st May to be included in 1.1? As #4985 shows they can cause saveloading problems. I haven't got much feedback on Empire buildings so far, and I'm afraid I won't be able to spend much time on Widelands for quite a while… :disappointed: I may still be able to do Barbarian and Atlantean buildings without adjusting scales if it is deemed important. Workers and critters are unlikely though.

(so this means no naval warfare (#855) in 1.1, and very unlikely #662 and #2058, right? also no hi-res work that need code changes, like #2087 #2089 and #1071?)

**Reply by Noordfrees** [reply]
Additional high-res graphics can always be included. Saveloading problems only occur when a scenario unit references files outside the unit's own folder, which should have been resolved for new saves by now. So we need to keep the old compat files until after v1.0 but any new saves should not have these problems.

All the long-awaited goodies like naval warfare, trading, 4k / UI scaling, scenario editor, shipyard stats and many more will have to wait another release cycle…

**Reply by hessenfarmer** [reply]
I definitly would like to see as much high res graphics for legacy units in 1.1 as possible. even at the cost of delaying the schedule for a month

**Reply by tothxa** [reply]
I don't think delaying the release will be necessary for buildings. I can't commit to anything with workers and critters though (but they may even be able to make it).

**Comment by tothxa** [comment]
Should we change all single frame animations to spritesheets for 1.1, to allow removing the unpacked animation code in 1.2?
(the high resolution work just for buildings of the old tribes already touches half of the affected init.luas anyway)

**Reply by Noordfrees** [reply]
We should keep support for non-packed animations around for add-ons backwards compatibility; it's not a very bug-prone or maintenance-heavy code path so we can afford to preserve it. The single-image animations should stay in the non-packed format then so we have testcases for the functionality in the official, non-modded game as well.

**Comment by hessenfarmer** [comment]
only a few bugs left we are getting there. Shall we have a tournament to test the RC? 
Having a tournament would probably get more testing. But it takes at least 4 weeks. maybe even 8 weeks.

**Reply by Noordfrees** [reply]
+1
As I mentioned [here](https://www.widelands.org/news/2022/Apr/22/widelands-1-1-first-snow-feature-freeze/#comment_anchor) I want to allow a long time for testing and eight weeks would not be too much.

And for the first time I'm going to participate :)

**Comment by hessenfarmer** [comment]
Ok tournament will start on september 5th.
see the homepage for details. 
more players would be highly appreciated

**Comment by Noordfrees** [comment]
# Release Schedule

The **release candidate** for Widelands 1.1 will be published on **Sunday, the 16th of October 2022**.
The **final release** shall follow one week later on **Sunday, the 23rd of October 2022**.

---

Another reminder: We still need lots of cool [**screenshots**](https://www.widelands.org/forum/topic/5571/) for the website and to refresh the appdata :)

**Comment by Noordfrees** [comment]
I'm unable to upload the binaries to the Launchpad milestone due to [this launchpad bug](https://answers.launchpad.net/launchpad/+question/698326), so I guess we might drop mirroring releases there and keep all artifacts on GitHub. This is unfortunate as LP offers support for checksums and signatures which GitHub lacks.

**Reply by matthiakl** [reply]
I don't know from where you take the release artifacts, but would it help the use the automated development action for this somehow? E.g. run on release branch and let it upload to 1.1 pre-release instead of `latest` tag. This also calculates md5 sums of the artifacts. But I guess manually calculating and uploading requires the same amount of time…

From the Release Draft:
> For Linux users, we provide a Flatpak package and a Ubuntu PPA and try to make compiling as easy as possible, and most distributions will pick up Widelands 1.1 into their repositories very quickly.

We also provide AppImages now, right?

**Reply by Noordfrees** [reply]
I updated the draft for the AppImages and also about supported MacOS configurations.

I prefer to do the binary uploads manually – I like to publish the release *after* the CI is done when all artifacts are available already, and also do some quick checks that the assets are valid and seem to work.
